### PR TITLE
WIP: Upgrade to 8.17.0

### PR DESCRIPTION
**Pull request by solidnerd**
I will continue the work from @iamchanghyunpark. Thanks for that :+1: . I need now to verify if it's correct implemented.

### CODE REVIEW

**Comment by solidnerd** [comment]
- [x] Build
- [x] Deploy
- [x] Expose necessary pages arguments
- [ ] Add Nginx Configuration for GitLab Pages
- [ ] Test this feature if it works

File: assets/runtime/functions [hunk 1]
```diff
@@ -1315,6 +1336,27 @@ configure_gitlab_shell() {
...
+  -pages-domain ${GITLAB_PAGES_DOMAIN}
+  -pages-root ${GITLAB_PAGES_DIR}
+  -listen-proxy :8090
+user=git
+autostart=true
+autorestart=true
+stdout_logfile=${GITLAB_INSTALL_DIR}/log/%(program_name)s.log
+stderr_logfile=${GITLAB_INSTALL_DIR}/log/%(program_name)s.log
+EOF
+gitlab_configure_pages
```
Comment (IlyaSemenov) [inline assets/runtime/functions]: Shouldn't this be indented? Also, the empty line below seems to be out of order (it should follow `}` not precede it).

Comment (jostyee) [inline assets/runtime/functions]: stderr logfile should be separate out with stdout.

**Comment by kristiandrucker** [comment]
Could you set gitlab version to 8.17.2? It's working for me from `upgrade-to-8-17-0` branch

**Comment by solidnerd** [comment]
@kristiandrucker  Did you test the pages feature at the moment I can't get it to work.

**Comment by kristiandrucker** [comment]
@solidnerd No I didn't

**Comment by kristiandrucker** [comment]
@solidnerd Is there any way I can see the logs for pages?

**Comment by kristiandrucker** [comment]
@solidnerd I get redirected to gitlab instance instead of my page.

**Comment by lenovouser** [comment]
Can Pages also be available in a different container?

**Comment by kristiandrucker** [comment]
@lenovouser Yeah that would be nice

**Comment by reynold-lariza** [comment]
FYI guys,

CURRENT ISSUE:
I'm currently using 8.17.0, and there seems to be a problem with commits not updating the merge request, even though the commit is seen from the source branch page.

And you cant downgrade anymore to 8.16.6, otherwise you'll get something like:

```
ActionView::Template::Error (Mysql2::Error: Unknown column 'timelogs.trackable_id' in 'where clause': SELECT SUM(`timelogs`.`time_spent`) FROM `timelogs` WHERE `timelogs`.`trackable_id` = 734 AND `timelogs`.`trackable_type` = 'MergeRequest'):
```

**Comment by jostyee** [comment]
@reynold-lariza I've upgraded to 8.17.2 w/o pages & everything is fine till now.

**Comment by reynold-lariza** [comment]
Hi @jostyee,

Have you tried adding commits to existing MR? Mine's not showing the additional commits :/

Also, does you gitlab ci builds works fine? I mean, does it auto run your builds upon update to target branch? or upon creation of tag (depends on gitlab-ci.yml `only` tag)

**Comment by jostyee** [comment]
@reynold-lariza We have a convention that do rebasing before resubmitting a MR & it works.
I've no idea whether the builtin CI is working or not, since it only support pull codes via HTTPS(we only allow SSH), so I build a Jenkins job & invoke it via webhook, working too.

**Comment by reynold-lariza** [comment]
@jostyee,

Yep, its fine pulling from https/ssh, however because we use built-in gitlab CI, so that's why I'm a bit frustrated, looks like I will have no choice but to inform the entire dev team and rollback sometime tomorrow :(

**Comment by kristiandrucker** [comment]
Hi, cannt you just create another container that will have volumes with gitlab pages data and the config for pages (for nginx) so it will be separate. It will be simpler for me and everyone running it behind Traefik or any proxy.

**Comment by reynold-lariza** [comment]
@jostyee,

Mate, it does not matter if we follow the same git flow or not. The context is that I'm expecting the same behavior similar to Github, and Stash, that if a commit is pushed to a source branch to an MR (or PR) it should update it the list of commits. If that is not the expected behavior, it shouldn't even allow it to be merged in the first place. 

Anyway, the reason I'm posting my observations is that everyone who are using Gitlab, know what others like you and me are experience during this update, right mate? ;)

**Comment by jostyee** [comment]
@reynold-lariza Fine here.
screen shot 2017-03-08 at 10 04 35 pm

**Comment by reynold-lariza** [comment]
Thanks @jostyee,

For now I've reverted from 8.17.0 to 8.16.6 manually (geez! what a pain! :D ), I'm currently observing that my existing project ci build are not working, but when I created a fresh project, gitlab ci is triggered. Must be something with after I upgraded from 8.16.6 to 8.17.0.... hmn... For now I'm slowly isolating, and upgrading very slowly, to figure out what happened to our instance :)

**Comment by Ilya-Kuchaev** [comment]
Any chance to see it released soon? :)

**Comment by kristiandrucker** [comment]
@solidnerd Could you please update gitlab version to 8.17.2? It works on my machine and it's the latest release. 
image

**Comment by jostyee** [comment]
@kristiandrucker why not upgrade GitLab Shell too if you want to use the latest version?

**Comment by kristiandrucker** [comment]
@jostyee Then what is the latest version?

**Comment by kristiandrucker** [comment]
Gitlab works on these versions from upgrade-to-8.17.0 branch. 
image

**Comment by paolomainardi** [comment]
@solidnerd do you need any kind of support ?

**Comment by reynold-lariza** [comment]
Hahaha, this is the toughest PR so far eh? :D

I had to revert ours for now to stable 8.16.6, I must be doing something wrong or incomplete that it messed up the DB. 

For now, I'll wait until this PR is merged ;)

**Comment by paolomainardi** [comment]
any kind of progress or something we can do to speed up the things ?

**Comment by solidnerd** [comment]
I would prefer someone that can test this with gitlab pages in a single docker-compose yml. I hope that i find some time this week to do it.

**Comment by kristiandrucker** [comment]
@solidnerd I will look into pages and how I can hook it up with a separate nginx container

**Comment by paolomainardi** [comment]
Just 7 days to gitlab 9.0 :)

**Comment by kristiandrucker** [comment]
@solidnerd @paolomainardi can you look at this nginx config provided here?

https://github.com/gitlabhq/omnibus-gitlab/blob/master/doc/settings/nginx.md#vhost-server-block

**Comment by kristiandrucker** [comment]
Nothing there. Dunno where is the nginx config.

**Comment by phpb-com** [comment]
Since so far this project is taking a break, I have span up my own fork https://github.com/phpb-com/docker-gitlab 

Hoping that it is only temporary ...

**Comment by kristiandrucker** [comment]
@phpb-com Can you look at official gitlab-ce and try to add mysql into it? Then it should be similar as this project. Official Gitlab-CE configures nginx depending on it's need. Pages should work then. Also then it will be good to setup auto rebuild when parent image is updated on docker hub to auto rebuild the image when gitlab will update theirs. Or look into Omnibus and "reverse engineer" it to have mysql support on CE Gitlab.

**Comment by solidnerd** [comment]
For everyone who is interesseted in this I will release every thing from 8.17 during the day. Pages works like excepted I could verfiy it.

**Comment by mikehaertl** [comment]
@solidnerd Thanks for your work on this, much appreciated. I assume that after merging this it's also easier to upgrade to 8.17.3, right? Any plans in this regard (and maybe even for 9.0)?

**Comment by solidnerd** [comment]
@mikehaertl Why is it hard to upgrade to `8.17.3` ? Are there any strange circumstances ?

**Comment by mikehaertl** [comment]
@solidnerd No no, I was just wondering because 8.17.0 took a little longer. I see you already released 8.17.3 now. Thanks!

**Pull request merged**